cppcache/integration/test/CachingProxyTest.cpp

```diff
@@ -51,86 +53,67 @@ class CachingProxyTest : public ::testing::Test {
 
   ~CachingProxyTest() override = default;
 
-  void SetUp() override {}
+  void SetUp() override {
+    cache.getPoolManager()
+        .createFactory()
+        .addLocator("localhost", cluster.getLocatorPort())
+        .create("pool");
+
+    region = cache.createRegionFactory(RegionShortcut::CACHING_PROXY)
+                 .setPoolName("pool")
+                 .create("region");
+  }
 
   void TearDown() override {}
 
   Cluster cluster = Cluster{LocatorCount{1}, ServerCount{1}};
+  Cache cache = CacheFactory().create();
```

Comment:
Correct. The default CTOR is a deleted function.

Reply:
Not sure what feels dirtier. Calling a builder pattern in the initializer for a member variable or converting that variable to `std::shared_ptr` and initializing it in the `SetUp`.

We should probably improve the `Cache` to default construct to an effectively "closed" cache so you can have it declared as a member but initialized to something useful later via move.

Reply:
That feels a bit contrived. Maybe we should just follow the pattern in the other new framework tests, which setup server and cache in each test.

Reply:
Seconded on wanting an explanation; nothing seems dirty here, this functionality was introduced in C++11 (though I personally don't care for it AT ALL, I can't find fault with it so I won't otherwise contest it).

Reply:
@mreddington Two issues:

1. We are initializing test specific objects in the fixture's, these will not be per-test method instances of these objects.
2. We are performing complex and exception throwing initialization of a member variable in the initialization of a type before the constructor is invoked. This particular factory can take variable parameters, like connection time outs, which could not be forwarded from some constructor or member method on this type. Initialization of a variable should never be done like this in production code so it also shouldn't in test code.

What this attempt to use the test fixture setup method has shown us it that we have a deficiency in how we defined the construction of `Cache` in that you can't declare it without also initializing it with some complex builder. Like many other state holding types in C++11 it should default construct to some safe but unusable state for its declaration, like `std::shared_ptr` default initializes to `nullptr`. It should then allow rvalue assignment from the value returned from the factory so that it can be fully initialized in the `SetUp` method. Similar to how someone with a `Application` class would declare it as a member and then initialize it someone in the application lifecycle.

Reply:
Based on the complexity of this change though @mmartell and I discussed holding off on using a test fixture to initialize these common values and copy the pattern of the other tests. We are then going to explore changes to `Cache` default construction and move assignment under another ticket.

Reply:
Switched to use TEST instead of TEST_F.